vividus-plugin-rest-api/src/main/java/org/vividus/archive/steps/ArchiveSteps.java

```diff
@@ -106,22 +112,40 @@ public void saveArchiveEntriesToVariables(DataWrapper archiveData, List<ArchiveV
     public void verifyArchiveContainsEntries(DataWrapper archiveData, List<NamedEntry> parameters)
     {
         Set<String> entryNames = ZipUtils.readZipEntryNamesFromBytes(archiveData.getBytes());
-
+        MutableBoolean isNeedToLoadAttachment = new MutableBoolean(false);
         parameters.forEach(entry ->
         {
             String expectedName = entry.getName();
             if (entry.getRule() != null)
             {
                 StringComparisonRule comparisonRule = entry.getRule();
-                softAssert.assertThat(String.format(
-                        "The archive contains entry matching the comparison rule '%s' with name pattern '%s'",
-                        comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)));
+                if (!softAssert.assertThat(String.format(
+                        "The archive contains entry matching the comparison "
+                        + "rule '%s' with name pattern '%s'", comparisonRule, expectedName), entryNames,
+                        hasItem(comparisonRule.createMatcher(expectedName))))
+                {
+                    isNeedToLoadAttachment.setValue(true);
+                }
             }
             else
             {
-                softAssert.assertThat("The archive contains entry with name " + expectedName, entryNames,
-                        hasItem(expectedName));
+                if (!softAssert.assertThat("The archive contains entry with name " + expectedName,
+                        entryNames, hasItem(expectedName)))
+                {
+                    isNeedToLoadAttachment.setValue(true);
+                }
             }
         });
+        if (entryNames.isEmpty())
+        {
+            attachmentPublisher.publishAttachment("There are no files in the archive"
+                    .getBytes(StandardCharsets.UTF_8), "emptyArchive.txt");
```

Comment:
txt??

Reply:
why not to handle this in archive-entries-result-table.ftl?

Reply:
what do we want as a result: empty file or file with line "There are no files in the archive" ?

Reply:
From my point of view the empty file is quite confusing

Reply:
ok, I'll try to handle empty archive in archive-entries-result-table.ftl with line `"There are no files in the archive"`

Reply:
done